verticapy/connection/connect.py

```diff
@@ -125,6 +129,52 @@ def connect(section: str, dsn: Optional[str] = None) -> None:
         prev_conn.close()
     try:
         gb_conn.set_connection(vertica_connection(section, dsn), section, dsn)
+        print("Connected Successfuly!")
+    except OAuthTokenRefreshError as e:
+        error_message = str(e)
+
+        from verticapy.connection.write import new_connection
+
+        # Check if the error message contains the refresh token part
+        if "Failed getting OAuth access token from a refresh token" in error_message:
+            # Handle the specific error related to the Refresh token part
+
+            print("Refresh Token or Client ID is not valid. Please re-enter")
+
+            new_connection(
+                conn_info=read_dsn(section, dsn), prompt=True, connect_attempt=False
+            )
+            try:
+                gb_conn.set_connection(vertica_connection(section, dsn), section, dsn)
+                print("Connected Successfuly!")
+            except OAuthTokenRefreshError as e:
+                print("Refresh token or Client ID is still not valid.")
+        else:
+            # Handle other types of ConnectionError
+            print("An error occurred:", error_message)
+
+    except ConnectionError as e:
+        error_message = str(e)
+        from verticapy.connection.write import new_connection
+
+        # Check if the error message contains the access token part
+        if "Token introspection failed" in error_message:
+            # Handle the specific error related to the Access token part
+            print("Access token is not valid. Please re-enter")
+
+            new_connection(
```

Comment:
why not creating a variable for the connection first and then passing to the function?